nordicsemi/dfu/dfu_transport_ble.py

```diff
@@ -163,34 +360,65 @@ def on_gap_evt_adv_report(self, ble_driver, conn_handle, peer_addr, rssi, adv_ty
+            self.conn_params = BLEGapConnParams(min_conn_interval_ms = 7.5,
+                                                max_conn_interval_ms = 30,
+                                                conn_sup_timeout_ms  = 4000,
+                                                slave_latency        = 0)
             logger.info('BLE: Found target advertiser, address: 0x{}, name: {}'.format(address_string, dev_name))
             logger.info('BLE: Connecting to 0x{}'.format(address_string))
-            self.adapter.connect(address = peer_addr, conn_params = conn_params)
+            self.adapter.connect(address = peer_addr, conn_params = self.conn_params)
             # store the address for subsequent connections
             self.target_device_addr = address_string
-
+            self.target_device_addr_type = peer_addr
 
     def on_notification(self, ble_adapter, conn_handle, uuid, data):
         if self.conn_handle         != conn_handle: return
-        if DFUAdapter.CP_UUID.value != uuid.value:  return
-        #logger.debug(data)
+        if DFUAdapter.CP_UUID.value != uuid.value:
+            return
         self.notifications_q.put(data)
 
+    def on_indication(self, ble_adapter, conn_handle, uuid, data):
+        if self.conn_handle         != conn_handle: return
+        if DFUAdapter.BLE_DFU_BUTTONLESS_BONDED_CHAR_UUID.value != uuid.value and \
+           DFUAdapter.BLE_DFU_BUTTONLESS_CHAR_UUID.value != uuid.value:
+            return
+        self.indication_q.put(data)
 
     def on_att_mtu_exchanged(self, ble_driver, conn_handle, att_mtu):
         logger.info('ATT MTU exchanged: conn_handle={} att_mtu={}'.format(conn_handle, att_mtu))
-
+        self.att_mtu = att_mtu
+        self.packet_size = att_mtu - 3
 
     def on_gattc_evt_exchange_mtu_rsp(self, ble_driver, conn_handle, **kwargs):
-        logger.info('ATT MTU exchange response: conn_handle={}'.format(conn_handle))
-    
+        pass
+
+
+class DfuBLEDriver(BLEDriver):
+    def __init__(self, serial_port, baud_rate=115200, auto_flash=False):
+        super(DfuBLEDriver, self).__init__(serial_port, baud_rate)
+
+    @NordicSemiErrorCheck
+    @wrapt.synchronized(BLEDriver.api_lock)
+    def ble_gap_sec_params_reply(self, conn_handle, sec_status, sec_params, own_keys, peer_keys):
+        assert isinstance(sec_status, BLEGapSecStatus), 'Invalid argument type'
+        assert isinstance(sec_params, (BLEGapSecParams, NoneType)), 'Invalid argument type'
+        assert isinstance(peer_keys, NoneType), 'NOT IMPLEMENTED'
+
+        self.__keyset = own_keys
+
+        return driver.sd_ble_gap_sec_params_reply(self.rpc_adapter,
+                                                  conn_handle,
+                                                  sec_status.value,
+                                                  sec_params.to_c() if sec_params else None,
+                                                  self.__keyset)
```

Comment:
Is there any reason why `self.__keyset` is set here, or could `own_keys` have been passed directly to this function?

Reply:
Yes, own_keys can probably be used here. The self.__keyset parameter was copied from the overridden method in ble_driver.py